lib/libxdp/libxdp.c

```diff
@@ -3253,8 +3297,14 @@ static int xdp_multiprog__attach(struct xdp_multiprog *old_mp,
 
 
 	err = xdp_attach_fd(prog_fd, old_fd, ifindex, mode);
-	if (err < 0)
+	if (err < 0) {
+		if (errno == EBUSY && !mp) {
+			pr_debug("Detaching link on ifindex %d%s\n", ifindex,
+			mode == XDP_MODE_SKB? " in skb mode":"");
```

Comment:
`bpf_link` is native-mode only, so instead of this we shouldn't attempt this at all if `mode == XDP_MODE_SKB`